### PR TITLE
Update CUDA-ExecutionProvider.md

### PR DESCRIPTION
**Pull request by jchen351**

### CODE REVIEW

File: docs/execution-providers/CUDA-ExecutionProvider.md [hunk 1]
```diff
@@ -38,6 +38,8 @@ ONNX Runtime built with cuDNN 8.x are not compatible with cuDNN 9.x.
 
 | ONNX Runtime             | CUDA   | cuDNN                                   | Notes                                                                                                                                                                                                                               |
 |--------------------------|--------|-----------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| 1.18                     | 12.4   | 10.0.1.6 (Linux)<br/>10.0.1.6 (Windows) | The default CUDA version for ORT 1.18.To install CUDA 12 package, please look at [Install ORT](../install). Java CUDA 12 support is back for release 1.18                                                                             |
```
Comment (sophies927) [inline docs/execution-providers/CUDA-ExecutionProvider.md:41]: This line makes it a bit confusing which CUDA version is the default for 1.18.

If it's CUDA 12, I think the notes column should say:

> The default CUDA version for ORT 1.18 is CUDA 12.4. To install CUDA 11 package, please look at [Install ORT](../install). Java CUDA 12 support is now available for release 1.18.

If it's CUDA 11.8, I think the notes column should say:

> The default CUDA version for ORT 1.18 is CUDA 11.8. To install CUDA 12 package, please look at [Install ORT](../install). Java CUDA 12 support is now available for release 1.18.

**Review by sophies927 — APPROVED** [review]
LGTM, thank you!

**Comment by omerferhatt** [comment]
Hi @sophies927. There's no version `10.0.1.6` for cuDNN. The latest version for cuDNN is [`9.1.1`](https://developer.nvidia.com/cudnn-downloads). There is some confusion between the cuDNN and TensorRT versions. Which is the last version for TensorRT is [`10.0.1.6`](https://github.com/NVIDIA/TensorRT/blob/release/10.0/VERSION)

**Comment by sophies927** [comment]
> Hi @sophies927. There's no version `10.0.1.6` for cuDNN. The latest version for cuDNN is [`9.1.1`](https://developer.nvidia.com/cudnn-downloads). There is some confusion between the cuDNN and TensorRT versions. Which is the last version for TensorRT is [`10.0.1.6`](https://github.com/NVIDIA/TensorRT/blob/release/10.0/VERSION)

@jchen351 can you please update with the correct cuDNN version?

**Pull request merged**